src/Commands/WaContent.php

```diff
@@ -0,0 +1,332 @@
+        if($id = $assocArgs['id'] ?? null) {
+            $this->import_composite($repository->find_by_id($id));
+        } else {
+            $repository->map_all(function($waContent){
+                $this->import_composite($waContent);
+            });
+        }
+    }
+
+    private function import_composite($waContent)
+    {
+        WP_CLI::line('Beginning import of: ' . $waContent->widget_content->title  . ' id: ' . $waContent->id);
+
+        $postId = $this->create_post($waContent);
+        $compositeContents = $this->format_composite_contents($waContent);
+
+        $this->handle_translation($postId, $waContent);
+        $this->set_meta($postId, $waContent);
+        $this->save_teasers($postId, $waContent);
+        $this->delete_orphaned_files($postId, $compositeContents);
+        $this->save_composite_contents($postId, $compositeContents);
+        $this->save_categories($postId, $waContent);
+        $this->save_tags($postId, $waContent);
+
+        WP_CLI::success('imported: ' . $waContent->widget_content->title  . ' id: ' . $waContent->id);
+    }
+
+    private function create_post($waContent)
+    {
+        $existingId = WpComposite::id_from_white_album_id($waContent->id);
+
+        // Tell Polylang the language of the post to allow multiple posts with the same slug in different languages
+        $_POST['term_lang_choice'] = $waContent->translation->locale;
+
+        $metaTitle = $waContent->widget_content->meta_title !== $waContent->widget_content->title ?
+            $waContent->widget_content->meta_title : null;
+        $metaDescription = $waContent->widget_content->meta_description !== $waContent->widget_content->description ?
+            $waContent->widget_content->meta_description : null;
+
+        return wp_insert_post([
+            'ID' => $existingId,
+            'post_title' => $waContent->widget_content->title,
+            'post_name' => $waContent->slug,
+            'post_status' => $waContent->widget_content->live ? 'publish' : 'draft',
+            'post_type' => WpComposite::POST_TYPE,
+            'post_date' => $waContent->widget_content->publish_at,
+            'post_modified' => $waContent->widget_content->publish_at,
+            'post_author' => $this->get_author($waContent),
+            'meta_input' => [
+                WpComposite::POST_META_WHITE_ALBUM_ID => $waContent->id,
+                WpComposite::POST_META_TITLE => $metaTitle,
+                WpComposite::POST_META_DESCRIPTION => $metaDescription,
+                WpComposite::POST_CANONICAL_URL => $waContent->widget_content->canonical_link,
+            ],
+        ]);
+
+    }
+
+    private function handle_translation($postId, $waContent)
+    {
+        pll_set_post_language($postId, $waContent->translation->locale); // Set post language
+    }
+
+    private function set_meta($postId, $waContent)
+    {
+        update_field('kind', 'Article', $postId); // Todo: set proper kind
```

Comment:
Should you set the proper kind?

Reply:
Not sure.. we are only importing articles and galleries, but in willow there is no longer any difference between the two..

Reply:
For now I will leave it at Gallery